lib/utils/isStandardSyntaxRule.js

```diff
@@ -37,6 +37,11 @@ module.exports = function (rule/*: Object*/)/*: boolean*/ {
     return false
   }
 
+  // Less guards
+  if ((/when.*?\(.*?\)/).test(selector)) {
```

Comment:
@davidtheclark in theory allow `when not not (@variable = true)` (does not make sense)